src/include/utils/minmax_and_dary_heap.h

```diff
@@ -0,0 +1,722 @@
+/**
+ * @file   fixed_min_heap.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Implementation of a min-max heap, taken from https://github.com/skarupke/heap
+ * Copyright Malte Skarupke 2020.
+ * Distributed under the Boost Software License, Version 1.0.
+ * (See http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+
+#ifndef TILEDB_MIN_MAX_HEAP_H
+#define TILEDB_MIN_MAX_HEAP_H
```

Comment:
I don't see anything referencing this file - am I just missing the use of it? Or do we not need to add it?

Would also be good to add a few tests.

Reply:
I was experimenting with it in `greedy_search` and `best_first`.  The annoying thing is that if you keep your frontier in a priority queue, you need to access both the min element and the max element at different points in the algorithm. However, I realized at some point that the queue doesn't have to be materialized and so we don't actually need the `min_max_heap`.  Deleted.

Reply:
Yes.